testplan/runnable/interactive/base.py

```diff
@@ -303,11 +325,19 @@ def run_test_suite(self, test_uid, suite_uid, await_results=True):
             self.logger.debug('Run suite ["%s" / "%s"]', test_uid, suite_uid)
             self._update_reports(
                 self.test(test_uid).run_testcases_iter(
-                    testsuite_pattern=suite_uid
+                    testsuite_pattern=suite_uid,
+                    suites_cases=suites_cases,
```

Comment:
I would that not to pass suites_cases down to Test level object, but to expand that here and call run_testcases_iter multiple times with testsuite_pattern & testcase_pattern.

Reply:
I checked this logic. We would repeatedly generate the test context because the property rather dynamic. It has a quadratic complexity so I would be in favor of keeping the suites_cases.